derive/src/utils.rs

```diff
@@ -112,21 +142,13 @@ pub fn variant_index(v: &Variant, i: usize) -> TokenStream {
 		None
 	});
 
-	// if no attribute, we fall back to an explicit discriminant (ie 'enum A { Foo = 1u32 }').
-	if index.is_none() {
-		if let Some((_, Expr::Lit(ExprLit { lit: Lit::Int(disc_lit), .. }))) = &v.discriminant {
-			index = disc_lit.base10_parse::<usize>().ok()
-		}
-	}
-
-	// fall back to the variant index if no attribute or explicit discriminant is found.
-	let index = index.unwrap_or(i);
-
-	// output the index with no suffix (ie 1 rather than 1usize) so that these can be quoted into
-	// an array of usizes and will work regardless of what type the discriminant values actually
-	// are.
-	let s = proc_macro2::Literal::usize_unsuffixed(index);
-	quote! { #s }
+	// then fallback to discriminant or just index
+	index.map(|i| quote! { #i }).unwrap_or_else(|| {
+		v.discriminant
+			.as_ref()
+			.map(|(_, expr)| quote! { (#expr) as isize})
```

Comment:
I wrote this so it compiles, but it is probably wrong. If the enum is `repr(u64)` it doesn't fit into isize